test/ratelimeter.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+var should = require('should');
+var sinon = require('sinon');
+
+var RateLimiter = require('../lib/ratelimiter');
+
+describe('RateLimiter', function() {
+
+  describe('@constructor', function() {
+    it('will instantiate without options', function() {
+      var limiter = new RateLimiter();
+      should.exist(limiter);
+    });
+    it('will instantiate without new', function() {
+      /* jshint newcap:false */
+      var limiter = RateLimiter();
+      should.exist(limiter);
+    });
+    it('will instantiate with options', function() {
+      var whitelist = [];
+      var blacklist = [];
+      var node = {};
+      var limiter = new RateLimiter({
+        node: node,
+        whitelist: whitelist,
+        blacklist: blacklist,
+        limit: 1,
+        interval: 1,
+        whitelistLimit: 1,
+        whitelistInterval: 1,
+        blacklistLimit: 1,
+        blacklistInterval: 1
+      });
+      should.exist(limiter);
+      should.exist(limiter.config);
+      should.exist(limiter.clients);
+      should.exist(limiter.node);
+      limiter.whitelist.should.equal(whitelist);
+      limiter.blacklist.should.equal(blacklist);
+      limiter.config.whitelist.totalRequests.should.equal(1);
+      limiter.config.whitelist.interval.should.equal(1);
+      limiter.config.blacklist.totalRequests.should.equal(1);
+      limiter.config.blacklist.interval.should.equal(1);
+      limiter.config.normal.interval.should.equal(1);
+      limiter.config.normal.totalRequests.should.equal(1);
+    });
+  });
+
+  describe('#middleware', function() {
```

Comment:
Tests added in: https://github.com/bitpay/insight-api/pull/467/commits/ae9f6a2c2f05b546f8c6cd2258f2478b78528da0